src/main/java/no/ks/fiks/io/klient/FiksIOUtsendingKlient.java

```diff
@@ -46,35 +48,62 @@ public static FiksIOUtsendingKlientBuilder builder() {
     }
 
     public SendtMeldingApiModel send(@NonNull MeldingSpesifikasjonApiModel metadata, @NonNull Optional<InputStream> data) {
-        try (final HttpEntity httpEntity = createMultiPartContent(metadata, data)) {
-            final ClassicHttpRequest request = requestFactory.createSendToFiksIORequest(httpEntity);
-            authenticationStrategy.setAuthenticationHeaders(request);
-            requestInterceptor.apply(request);
-            return client.execute(request, response -> {
-                final int responseCode = response.getCode();
-                log.debug("Response status: {}", responseCode);
-                try (final HttpEntity entity = response.getEntity();
-                     final InputStream content = entity.getContent()) {
-                    if (responseCode >= HttpStatus.SC_BAD_REQUEST) {
-                        final var contentAsString = IOUtils.toString(content, StandardCharsets.UTF_8);
-                        throw new FiksIOHttpException(String.format("HTTP-feil under sending av melding (%d): %s", responseCode, contentAsString), responseCode, contentAsString);
-                    }
-                    return objectMapper.readValue(content, SendtMeldingApiModel.class);
-                }
-            });
+        if (data.isPresent()) {
+            return send(metadata, data.get());
+        } else {
+            return send(metadata);
+        }
+    }
+    private SendtMeldingApiModel send(@NonNull MeldingSpesifikasjonApiModel metadata) {
+        var multipartRequestContentBuilder = MultipartEntityBuilder.create()
+                .addBinaryBody("metadata", serialiser(metadata).getBytes(StandardCharsets.UTF_8), ContentType.APPLICATION_JSON, null);
+        HttpEntity httpEntity =  multipartRequestContentBuilder.build();
+        try {
+            return sendMelding(httpEntity);
         } catch (IOException e) {
-            throw new RuntimeException("Feil under invokering av FIKS IO api", e);
+            throw new RuntimeException("Feil under sending av Fiks-IO melding", e);
+        }
```

Comment:
Kunne catchen for IOException vært inne i sendMelding, siden den nå håndteres likt to steder?

Reply:
delvis, pga pis.read() også kan kaste IOException. Men flyttet alikevell. Det blir ryddigere og bedre feilmelding på hva som kastet ioexception